diff --git a/pommed/kbd_backlight.h b/pommed/kbd_backlight.h
index 0b40091..876e6ed 100644
--- a/pommed/kbd_backlight.h
+++ b/pommed/kbd_backlight.h
@@ -71,6 +71,7 @@ has_kbd_backlight(void)
 	  || (mops->type == MACHINE_MACBOOKPRO_6)
 	  || (mops->type == MACHINE_MACBOOKPRO_7)
 	  || (mops->type == MACHINE_MACBOOKPRO_8)
+	  || (mops->type == MACHINE_MACBOOKPRO_9)
 	  || (mops->type == MACHINE_MACBOOK_5)
 	  || (mops->type == MACHINE_MACBOOKAIR_1)
 	  || (mops->type == MACHINE_MACBOOKAIR_2));
diff --git a/pommed/pommed.c b/pommed/pommed.c
index 6916eec..9297acf 100644
--- a/pommed/pommed.c
+++ b/pommed/pommed.c
@@ -315,6 +315,14 @@ struct machine_ops mb_mops[] = {
     /* .evdev_identify = evdev_is_wellspring5, */
   },
 
+  {  /* MacBookPro9.2 (15", Mid 2012) */
+    .type = MACHINE_MACBOOKPRO_9,
+    .lcd_backlight_probe = mbp_sysfs_backlight_probe,
+    .lcd_backlight_step = sysfs_backlight_step,
+    .lcd_backlight_toggle = sysfs_backlight_toggle,
+    /* .evdev_identify = evdev_is_wellspring5, */
+  },
+
   /* MacBook machines */
 
   {  /* MacBook1,1 (Core Duo) */
@@ -756,6 +764,9 @@ check_machine_dmi(void)
 	   || (strcmp(buf, "MacBookPro8,2") == 0)
 	   || (strcmp(buf, "MacBookPro8,3") == 0))
     ret = MACHINE_MACBOOKPRO_8;
+  /* MacBook Pro 15" (Mid 2012) */
+  else if (strcmp(buf, "MacBookPro9,2") == 0)
+    ret = MACHINE_MACBOOKPRO_9;
   /* Core Duo MacBook (May 2006) */
   else if (strcmp(buf, "MacBook1,1") == 0)
     ret = MACHINE_MACBOOK_1;
diff --git a/pommed/pommed.h b/pommed/pommed.h
index 3172358..930b57d 100644
--- a/pommed/pommed.h
+++ b/pommed/pommed.h
@@ -38,6 +38,7 @@ typedef enum
     MACHINE_MACBOOKPRO_6,
     MACHINE_MACBOOKPRO_7,
     MACHINE_MACBOOKPRO_8,
+    MACHINE_MACBOOKPRO_9,
 
     MACHINE_MACBOOK_1,
     MACHINE_MACBOOK_2,
